### DISCUSSION TITLE
[Question]: Is it possible to add Filter Settings to Gemini API?

### DISCUSSION THREAD

**Discussion by isandom** [danny-avila/LibreChat#3531] [Q&A]
### What is your question?

Filter settings are currently available for Vertex AI API in LibreChat, but it's not working for the **Gemini API**.

apparently they can also be enabled for Gemini API through the API key:
https://ai.google.dev/gemini-api/docs/safety-settings

### More Details

**Vertex AI API** has had this limitation for a while but it should be enabled for the **Gemini API**

With this implementation we should be able to use the "BLOCK_NONE" filter for general use via the **Gemini API** instead of getting the error message in LibreChat.

It can be done according to this reddit post: https://www.reddit.com/r/googlecloud/comments/1d22h2q/how_to_use_block_none_with_gemini_api_after_new/

> the Non-Vertex Gemini API (the one with the Free Tier that they claimed would be 60 rpm and then quietly reduced it to 15 RPM without any announcement or warning at all), **does** still allow you to use BLOCK_NONE.

Is this possible to implement or am I missing something?

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lidonius1122** [comment]
This PR should fix this.

https://github.com/danny-avila/LibreChat/pull/3533
